unittests/list_restriction/test_list_restriction.cpp

```diff
@@ -0,0 +1,358 @@
+                                             "         <soap:body"
+                                             "            encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\""
+                                             "            namespace=\"urn:examples:helloservice\""
+                                             "            use=\"encoded\"/>"
+                                             "      </input>"
+                                             "      <output>"
+                                             "         <soap:body"
+                                             "            encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\""
+                                             "            namespace=\"urn:examples:helloservice\""
+                                             "            use=\"encoded\"/>"
+                                             "      </output>"
+                                             "   </operation>"
+                                             "   </binding> "
+                                             "   <service name=\"Hello_Service\">"
+                                             "      <documentation>WSDL File for HelloService</documentation>"
+                                             "      <port binding=\"tns:Hello_Binding\" name=\"Hello_Port\">"
+                                             "         <soap:address"
+                                             "            location=\"http://www.examples.com/SayHello/\"/>"
+                                             "      </port>"
+                                             "   </service>"
+                                             "</definitions>");
+        QXmlInputSource source;
+        source.setData(srcXml);
+        QXmlSimpleReader reader;
+        reader.setFeature(QLatin1String("http://xml.org/sax/features/namespace-prefixes"), true);
+
+        QString errorMsg;
+        int errorLine, errorCol;
+        QDomDocument doc;
+        doc.setContent(&source, &reader, &errorMsg, &errorLine, &errorCol);
+
+        QDomElement element = doc.documentElement();
+        NSManager namespaceManager;
+        namespaceManager.setPrefix(QLatin1String("xml"), NSManager::xmlNamespace());
+
+        MessageHandler messageHandler;
+        ParserContext context;
+        context.setNamespaceManager(&namespaceManager);
+        context.setMessageHandler(&messageHandler);
+
+        KWSDL::Definitions definitions;
+        if (definitions.loadXML(&context, element)) {
+            definitions.fixUpDefinitions(/*&context, element*/);
+            KODE::Code::setDefaultIndentation(4);
+            KWSDL::WSDL wsdl;
+            wsdl.setDefinitions(definitions);
+            wsdl.setNamespaceManager(namespaceManager);
+            KWSDL::Converter converter;
+            converter.setWSDL(wsdl);
+
+            converter.convert();
+            foreach (auto elem, converter.classes()) {
```

Comment:
All this test, until this point, is very similar to the previous one.

You could extract a helper function for all three tests, which takes XML as input, and returns a list of classes as output?
With qtestlib that's a bit tricky (QVERIFY/QCOMPARE inside the helper function requires that it returns void), there are several solutions. One solution would be to take `Class::List &` as second argument to the function (and assign to that parameter at the end of the function).